vllm/compilation/decorators.py

```diff
@@ -160,6 +163,11 @@ def __call__(self, *args, **kwargs):
         # compiled function and let torch.compile handle the dispatching,
         # with the overhead of guard evaluation and recompilation.
         if len(self.compiled_codes) < 1 or not self.use_custom_dispatcher:
+            # it seems Dynamo reuse the compilation across instances,
+            # while we need to make sure the compiled code is not reused.
+            # we need to control all the compilation of the model.
+            torch._dynamo.eval_frame.remove_from_cache(
```

Comment:
Why it is problematic to reuse compilation parts?

Reply:
say we have two llama 8B model, one wants to capture sizes 1, 2, and the other wants to capture sizes 3, 4. If they share the same compiled code, the second model will not see the profile run. the profile run is the secrect of our approach.